wagtail_newsletter/views.py

```diff
@@ -19,3 +22,48 @@ def recipients(request):
             "member_count": recipients.member_count,
         }
     )
+
+
+class NewsletterHistoryView(HistoryView):
+    object: NewsletterPageMixin
+
+    def get_object(self):
+        page = get_object_or_404(Page, id=self.pk).specific
+
+        if not isinstance(page, NewsletterPageMixin):
+            raise Http404
+
+        return page
+
+    def dispatch(self, request, *args, **kwargs):
+        if not self.object.has_newsletter_permission(
+            self.request.user, "save_campaign"
+        ):
+            return permission_denied(self.request)
+
+        return super().dispatch(request, *args, **kwargs)
+
+    # XXX Wagtail 5.2 expects a `model` property to generate the message
```

Comment:
I've called out the backwards compatibility hacks so that we can easily remove them as we drop support for old versions.